mindmeld/active_learning/data_loading.py

```diff
@@ -316,21 +352,25 @@ def sample_and_update(
         return newly_sampled_queries_ids
 
     @staticmethod
-    def filter_queries_by_domain(query_list: ProcessedQueryList, domain: str):
+    def filter_queries_by_nlp_component(
+        query_list: ProcessedQueryList, component_type: str, component_name: str
+    ):
         """Filter queries for training preperation.
 
         Args:
             query_list (list): List of queries to filter
-            domain (str): Domain of desired queries
+            component_type (str): Component type of desired queries
+            component_name (str): Component name of desired queries
 
         Returns:
             filtered_queries_indices (list): List of indices of filtered queries.
             filtered_queries (list): List of filtered queries.
         """
+
         filtered_queries = []
         filtered_queries_indices = []
         for index, query in enumerate(query_list.processed_queries()):
-            if query.domain == domain:
+            if getattr(query, component_type) == component_name:
```

Comment:
Clever use of getattr()! :) Consider throwing an error as an "else" condition if the component type is not found in the query.